python/lsst/resources/_resourcePath.py

```diff
@@ -1153,6 +1155,48 @@ def remove(self) -> None:
         """Remove the resource."""
         raise NotImplementedError()
 
+    @classmethod
+    def mremove(cls, uris: Iterable[ResourcePath]) -> None:
```

Comment:
In theory we'd like to return some indication as to what worked. For the futures implementation we could wait for completion and then raise an exception group for the failed deletes. The problem is that `delete_objects` seems to return without error if the file key doesn't exist. It does return an `Errors` list of keys that could not be removed due to other reasons but they are mostly reasons that lead to everything failing. That might be okay though.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/delete_objects.html